addons/godot-xr-tools/xr/start_xr.gd

```diff
@@ -0,0 +1,275 @@
+var xr_interface : ARVRInterface
+
+## XR active flag
+var xr_active : bool = false
+
+# OpenXR configuration (of type OpenXRConfig.gdns)
+var _openxr_configuration
+
+# OpenXR enabled extensions
+var _openxr_enabled_extensions : Array
+
+
+# Handle auto-initialization when ready
+func _ready() -> void:
+	if !Engine.editor_hint and auto_initialize:
+		initialize()
+
+
+## Initialize the XR interface
+func initialize() -> bool:
+	# Check for OpenXR interface
+	xr_interface = ARVRServer.find_interface('OpenXR')
+	if xr_interface:
+		_setup_for_openxr()
+		return true
+
+	# Check for WebXR interface
+	xr_interface = ARVRServer.find_interface('WebXR')
+	if xr_interface:
+		_setup_for_webxr()
+		return true
+
+	# No XR interface
+	xr_interface = null
+	print("No XR interface detected")
+	return false
+
+
+# Check for configuration issues
+func _get_configuration_warning():
+	if physics_rate_multiplier < 1.0:
+		return "Physics rate multiplier should be at least 1x the HMD rate"
+
+	return ""
+
+
+# Perform OpenXR setup
+func _setup_for_openxr() -> void:
+	print("OpenXR: Configuring interface")
+
+	# Load the OpenXR configuration resource
+	var openxr_config_res := load("res://addons/godot-openxr/config/OpenXRConfig.gdns")
+	if not openxr_config_res:
+		push_error("OpenXR: Unable to load OpenXRConfig.gdns")
+		return
+
+	# Create the OpenXR configuration class
+	_openxr_configuration = openxr_config_res.new()
+
+	# Initialize the OpenXR interface
+	if not xr_interface.interface_is_initialized:
+		print("OpenXR: Initializing interface")
+		if not xr_interface.initialize():
+			push_error("OpenXR: Failed to initialize")
+			return
+
+	# Connect the OpenXR events
+	ARVRServer.connect("openxr_session_begun", self, "_on_openxr_session_begun")
+	ARVRServer.connect("openxr_visible_state", self, "_on_openxr_visible_state")
+	ARVRServer.connect("openxr_focused_state", self, "_on_openxr_focused_state")
+
+	# Read the OpenXR enabled extensions
+	_openxr_enabled_extensions = _openxr_configuration.get_enabled_extensions()
+
+	# Check for passthrough
+	if enable_passthrough and _openxr_is_passthrough_supported():
+		enable_passthrough = _openxr_start_passthrough()
+
+	# Switch the viewport to XR
+	get_viewport().arvr = true
+
+
+# Handle OpenXR session ready
+func _on_openxr_session_begun() -> void:
+	print("OpenXR: Session begun")
+
+	# Our interface will tell us whether we should keep our render buffer in linear color space
+	get_viewport().keep_3d_linear = _openxr_configuration.keep_3d_linear()
+
+	# increase our physics engine update speed
+	var refresh_rate : float = _openxr_configuration.get_refresh_rate()
+	if refresh_rate > 0:
+		# Report provided frame rare
+		print("OpenXR: HMD refresh rate is set to ", str(refresh_rate))
+	else:
+		# None provided, assume a standard rate
+		print("OpenXR: No refresh rate given by XR runtime")
+		refresh_rate = 144
+
+	# Pick a physics rate
+	var physics_rate := int(refresh_rate * physics_rate_multiplier + 0.5)
```

Comment:
`+0.5`?

Reply:
It's a fast way of rounding positive values to integer as casting to int does a floor operation; however for clarity I've changed it to int(round(<expression>))